src/components/BlogPosts.jsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import { blogPosts } from "../constants";
+import { AiFillGithub } from "react-icons/ai";
+import { BsLink45Deg } from "react-icons/bs";
+import { motion } from "framer-motion";
+import Button from "./Button";
+
+const BlogPost = (props) => {
+  return (
+    <motion.div
+      className="overflow-hidden transition-colors duration-300 transform border rounded-xl hover:border-transparent group dark:border-gray-700 dark:hover:border-transparent feature-card"
+      whileInView={{ y: [-40, 0], opacity: [0, 1] }}
+      transition={{ duration: 1 }}
+    >
+      <img
+        className="flex-shrink-0 object-cover w-full h-40"
+        src={props.image}
+        alt=""
+      />
+      <div className="px-12 py-8 ">
+        <div className="flex flex-col">
+          <div>
+            <p className="font-poppins font-normal text-dimWhite mb-3">
+              {props.date}
+            </p>
+            <h1
+              className="text-xl font-semibold font-poppins text-gray-700 capitalize md:text-2xl group-hover:text-white text-gradient blog-title"
+              title={props.title}
+            >
+              {props.title}
+            </h1>
+            <p className="mt-5 text-gray-500 capitalize dark:text-gray-300 group-hover:text-gray-300">
+              <div className="flex sm:flex-row">
+                {props.tags.map((tag, index) => (
+                  <div
+                    key={tag.id}
+                    index={index}
+                    className="text-dimWhite mr-5 text-sm hover:text-teal-200 p-1.5 ring-1 ring-dimWhite hover:ring-teal-200 rounded text-ellipsis whitespace-nowrap overflow-hidden"
+                    title={tag.name}
+                  >
+                    <span className="cursor-default">{tag.name}</span>
+                  </div>
+                ))}
+              </div>
+            </p>
+          </div>
+        </div>
+
+        <div className="mt-8">
+          <a href={props.link} target="_blank">
+            <Button styles="" text="Read more &rarr;" icon="AiFillGithub" />

```

Comment:
Do we need the GitHub icon here?

Reply:
Hello, it was needed otherwise the Button component would break. It has been fixed in [a424987](https://github.com/mittal-parth/personal-portfolio/pull/40/commits/a4249877399aa44b093c7d492148d80a78101b31)
Please take a look at the commit message for more details

Reply:
Ohk, thanks for the detailed commit message